### PR TITLE
Add RateLimitPolicy create page

### LINKED ISSUE #31: Rate limit policy create/update
WHAT:
Support for creating and updating Kuadrant policies (RLP) via the UI, with appropriate forms.

HOW:

Examples and scaffolding are already in place that we can use to do this. (Without Picker)
For updating we should be using the view given from creation greying out any info that cannot be updated

Mockup discussed with UI are what we want the end goal to look like.
Call recording: https://drive.google.com/file/d/1YG_nmMuB6hgvL-bnciKgvbdjO3DBhcaF/view?usp=drive_web
Mockup: https://www.figma.com/proto/r6rh1Jkx9vI9G2pXv0rT2D/OCP-Dynamic-plugin-(GA)?page-id=0%3A1&node-id=42-5365&viewport=1101%2C329%2C0.39&t=1Z0YW11ZmJEPrOJX-1&scaling=min-zoom&content-scaling=fixed&starting-point-node-id=21%3A8212

In the Administrator perspective, users should be able to create all types of policy
In the Developer perspective, users should be able to delete: AuthPolicy, RateLimitPolicy

### PR DESCRIPTION
**Pull request by david-martin**
Closes #31 

Screenshots below.
A few fields have been purposefully left out of the form view as the API will be changing for kuadrant v1.
Future changes to the from are captured in https://github.com/Kuadrant/console-plugin/issues/68
In the interim, an info message is shown to direct the user to the yaml view for more complex limits configuration beyond simple limits.

### Create form

image

### Add Limit modal

image

### Limits in form

image

### Yaml editor

image

### CODE REVIEW

File: src/components/KuadrantRateLimitPolicyCreatePage.tsx [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+    // Reset temporary state when opening modal for new entry
+    setNewLimit({ rates: [{ duration: 60, limit: 100, unit: 'minute' }] });
+    setRateName('');
+    setIsModalOpen(true);
+  };
+  const handleCloseModal = () => setIsModalOpen(false);
+  const handleSave = () => {
+    if (!rateName) {
+      // Add validation if rateName is empty
+      alert('Limit Name is required!');
```
Comment (jasonmadigan) [inline src/components/KuadrantRateLimitPolicyCreatePage.tsx:67]: Maybe use `<Alert>`?

File: src/components/KuadrantRateLimitPolicyCreatePage.tsx [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+          group: 'gateway.networking.k8s.io',
+          kind: 'Gateway',
+          name: selectedGateway.name,
+          namespace: selectedGateway.namespace,
+        },
+        limits: { ...limits },
+      },
+    };
+
+    // setYamlResource(removeUndefinedFields(updatedYamlResource)); // Clean undefined values?
```
Comment (jasonmadigan) [inline src/components/KuadrantRateLimitPolicyCreatePage.tsx:156]: rm

File: src/components/KuadrantRateLimitPolicyCreatePage.tsx [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+                setNewLimit={setNewLimit}
+                rateName={rateName}
+                setRateName={setRateName}
+                handleSave={handleSave}
+              />
+            </div>
+            <Alert
+              variant="info"
+              isInline
+              title="To set defaults, overrides, and more complex limits, use the YAML view."
```
Comment (jasonmadigan) [inline src/components/KuadrantRateLimitPolicyCreatePage.tsx:309]: think this is perfectly reasonable tbh

**Review by jasonmadigan — COMMENTED** [review]
few small things, but other than that, looks good.

**Comment by david-martin** [comment]
@jasonmadigan thanks for reviewing.
I've pushed up recommended changes.
Also added i18n strings.

**Comment by jasonmadigan** [comment]
Changes look good, let me give it a quick spin again

**Comment by jasonmadigan** [comment]
Works good, noticed one small thing - the edit action in the RLP list view doesn't seem to navigate to edit

Screenshot 2024-09-13 at 16 06 52

**Comment by david-martin** [comment]
> the edit action in the RLP list view doesn't seem to navigate to edit

I'd say that's the same for all Edit buttons?
There's a separate issue to figure out reusing the create view as an edit view #59

**Comment by jasonmadigan** [comment]
Must've had a fever dream where I thought some of these worked - nevermind so!

/lgtm

**Pull request merged**